15/solve.rb

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env ruby
+def total_score(ingredients, variant)
+  total_score = 1
+  (0..3).each do |ingredient|
+    property_sum = 0
+    variant.each_index do |index|
+      property_sum += ingredients[index][ingredient] * variant[index]
+    end
+    property_sum = property_sum > 0 ? property_sum : 0
+    total_score *= property_sum
+  end
+  total_score
+end
+
+def cal_500?(ingredients, variant)
```

Comment:
Imagine grumpy cat here. Do you really want to hardwire the calory limit this way? :)